ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/JarManifestErrorReporter.java

```diff
@@ -138,11 +138,21 @@ protected void parseManifest(IDocument document, IProgressMonitor monitor) {
 					return;
 				}
 				// parse
+				String rest = document.get().substring(lineInfo.getOffset() + lineInfo.getLength()).trim();
 				if (line.length() == 0) {
 					// Empty Line
 					if (l == 0) {
 						report(PDECoreMessages.BundleErrorReporter_noMainSection, 1, CompilerFlags.ERROR, PDEMarkerFactory.CAT_FATAL);
 						return;
+
+					} else if (rest != "") { //$NON-NLS-1$
+						VirtualMarker marker = report(PDECoreMessages.BundleErrorReporter_noNameHeader, lineNumber,
```

Comment:
thanks @iloveeclipse for the proposal. I had tried a similar approach earlier, but abandoned. My plan was to identify multiple continues empty lines and mark them together. The fact that the markers do not allow aggregations in their attributes (only accepts strings and integers as values), it was tricky to implement that.

However, your suggestion to `remember` empty lines and reporting at the end individually makes sense to me, I will follow that; thanks!

Reply:
@iloveeclipse  - pushed a new change based on the review comments, please have a look. The approach is to identify empty lines in each iteration and push the locations to a list instead of marking inline, and then mark those only when a non-empty line is visited. this solves two concerns: no pressure is incurred in the iteration, legitimate empty lines at the end are automatically ignored.